### ISSUE TITLE
Update-Module fails on each module (Cannot bind argument to parameter 'ProviderName' because it is an empty string.)

### ISSUE THREAD

**Issue by Manbearpiet** [PowerShell/PSResourceGet#1373]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/PSResourceGet)
- [X] Search the existing issues.

### Steps to reproduce

I am experiencing issues on updating and (un-)installing modules using PowerShellGet.

This happens with any module I try.

Using Pester as an Example. I also tried removing 3.0.20, to no avail.

### Expected behavior

```console
Update-Module Pester -Force

Install-Module Pester

to work like `Install-PSResource pester` producing no errors or warnings
```

### Actual behavior

```console
Update-Module Pester -Force
WARNING: Cannot bind argument to parameter 'fastPackageReference' because it is an empty string.
WARNING: Cannot bind argument to parameter 'ProviderName' because it is an empty string.
Install-Package: Cannot bind argument to parameter 'ProviderName' because it is an empty string.


Install-Module Pester
WARNING: Cannot bind argument to parameter 'fastPackageReference' because it is an empty string.
WARNING: Cannot bind argument to parameter 'ProviderName' because it is an empty string.
```

### Error details

```console
Exception             :
    Type    : System.Exception
    Message : Cannot bind argument to parameter 'ProviderName' because it is an empty string.
    HResult : -2146233088
TargetObject          : Microsoft.PowerShell.PackageManagement.Cmdlets.InstallPackage
CategoryInfo          : InvalidData: (Microsoft.PowerShel…lets.InstallPackage:InstallPackage) [Install-Package], Exception
FullyQualifiedErrorId : ParameterArgumentValidationErrorEmptyStringNotAllowed,New-SoftwareIdentityFromPSGetItemInfo,Microsoft.PowerShell.PackageManag
ement.Cmdlets.InstallPackage
InvocationInfo        :
    MyCommand        : Install-Package
    ScriptLineNumber : 13069
    OffsetInLine     : 20
    HistoryId        : 36
    ScriptName       : /usr/local/microsoft/powershell/7/Modules/PowerShellGet/PSModule.psm1
    Line             : $sid = PackageManagement\Install-Package @PSBoundParameters

    PositionMessage  : At /usr/local/microsoft/powershell/7/Modules/PowerShellGet/PSModule.psm1:13069 char:20
                       + …           $sid = PackageManagement\Install-Package @PSBoundParameters
                       +                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    PSScriptRoot     : /usr/local/microsoft/powershell/7/Modules/PowerShellGet
    PSCommandPath    : /usr/local/microsoft/powershell/7/Modules/PowerShellGet/PSModule.psm1
    InvocationName   : PackageManagement\Install-Package
    CommandOrigin    : Internal
ScriptStackTrace      : at Update-Module<Process>, /usr/local/microsoft/powershell/7/Modules/PowerShellGet/PSModule.psm1: line 13069
                        at <ScriptBlock>, <No file>: line 1
                        at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo :
```

### Environment data

```powershell
Get-Module Microsoft.PowerShell.PSResourceGet; $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.3.6
PSEdition                      Core
GitCommitId                    7.3.6
OS                             Darwin 22.6.0 Darwin Kernel Version 22.6.0: Wed Jul  5 22:22:05 PDT 2023; root:xnu-8796.141.3~6/RELEASE_ARM64_T6000
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

# Get-Module Microsoft.PowerShell.PSResourceGet returns nothing

gmo powershellget -li

    Directory: /Users/christianpiet/.local/share/powershell/Modules

ModuleType Version    PreRelease Name                                PSEdition ExportedCommands
---------- -------    ---------- ----                                --------- ----------------
Binary     3.0.21     beta21     PowerShellGet                       Desk      {Find-PSResource, Get-InstalledPSResource, Get-PSResourceRepository, …
Binary     3.0.19     beta19     PowerShellGet                       Desk      {Find-PSResource, Get-PSResource, Get-PSResourceRepository, Get-PSScr…
Binary     3.0.17     beta17     PowerShellGet                       Desk      {Find-PSResource, Get-PSResource, Get-PSResourceRepository, Install-P…
```

**Comment by LaurentDardenne** [comment]
Try to remove 'PowershellGet' module and use[ instead](https://devblogs.microsoft.com/powershell/powershellget-in-powershell-7-4-updates/) 'Microsoft.PowerShell.PSResourceGet'.

**Comment by Manbearpiet** [comment]
Installed the module, removed the 3.0.* versions of PowerShellGet. Should my workflow now use `Update-PSResource` to update modules? And `Install-PSResource` to install?

**Comment by LaurentDardenne** [comment]
>>Should my workflow now use Update-PSResource to update modules? And Install-PSResource to install?

It depends on your context, _'Microsoft.PowerShell.PSResource Get'_ is under development.

**Comment by Manbearpiet** [comment]
> > > Should my workflow now use Update-PSResource to update modules? And Install-PSResource to install?
> 
> It depends on your context, _'Microsoft.PowerShell.PSResource Get'_ is under development.

Ok, let me rephrase:

- Will `Update-PSResource` probably replace the Update operation now done with `Update-Module`?

- Will `Install-PSResource` probably replace the Install operation now done with `Install-Module`?

Ah I see: 

- [Install-PSResource](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.psresourceget/install-psresource?view=powershellget-3.x#description)

- [Update-PSResource](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.psresourceget/update-psresource?view=powershellget-3.x#description)

**Comment by Manbearpiet** [comment]
Thanks for your support @LaurentDardenne this fixed it 👍 🥳
